src/webview/MessageList.js

```diff
@@ -240,12 +201,35 @@ class MessageList extends Component<Props> {
     /**
      * Effective URL of the MessageList webview.
      *
+     * It points to `index.html` in the webview-assets folder. See
+     * `tools/build-webview` for more information on what folder that
+     * is.
+     *
      * We provide all HTML at creation (or refresh) time; the document named
      * here is not loaded, and doesn't even need to exist. It serves only as a
      * placeholder, so that relative URLs and cross-domain security restrictions
      * have somewhere to believe that this document originates from.
+     *
+     * - On iOS: We can't easily hardcode this because it includes
+     *   UUIDs. We grab it in some iOS native code and expose it
+     *   through the React Native bridge in ZLPConstants.m for this
+     *   purpose.
+     * - On Android: Different apps' WebViews see different (virtual)
+     *   root directories as `file:///`, and in particular the WebView
+     *   provides the APK's `assets/` directory as
+     *   `file:///android_asset/`. [1]  We can easily hardcode that,
+     *   so we do.
+     *
+     * [1] Oddly, this essential feature doesn't seem to be documented!  It's
+     *     widely described in how-tos across the web and StackOverflow answers.
+     *     It's assumed in some related docs which mention it in passing, and
+     *     treated matter-of-factly in some Chromium bug threads.  Details at:
+     *     https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/android.20filesystem/near/796440
      */
-    const baseUrl = `${assetsPath}/index.html`;
+    const baseUrl: WhatwgURL =
```

Comment:
This annotation is surely not needed :slightly_smiling_face: -- we're initializing it with `new WhatwgURL(…)`